### PR TITLE
GUI: Sync ButtonUi4 Spinner Fixes from LoginUI

### PR DESCRIPTION
**Pull request by Jon-edge**
image

### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [ ] Yes
- [x] No

### Dependencies

https://github.com/EdgeApp/edge-login-ui-rn/pull/157

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [x] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206455248718481

### CODE REVIEW

File: src/components/ui4/ButtonUi4.tsx [hunk 1]
```diff
@@ -196,9 +210,9 @@ const getStyles = cacheStyles((theme: Theme) => {
       // Reduce the bounds of a tertiary button so it doesn't appear to be too
```
Comment (paullinator) [inline src/components/ui4/ButtonUi4.tsx:210]: tertiary buttons should not have any difference in spacing. On some styles they may actually be filled with borders. Please create a task to fix in 4.1

File: src/components/ui4/ButtonUi4.tsx [hunk 2]
```diff
@@ -114,41 +113,56 @@ export function ButtonUi4(props: Props) {
...
   return (
-    <TouchableOpacity disabled={disabled || pending} style={finalContainerCommon} onPress={handlePress} testID={testID}>
+    <TouchableOpacity disabled={disabled || pending || spinner} style={finalContainerCommon} onPress={handlePress} testID={testID}>
       <LinearGradient
         {...gradientProps}
         style={[styles.contentCommon, dynamicGradientStyles, mini ? styles.contentSizeMini : styles.contentSizeDefault, ...finalContainerCommon]}
       >
-        {hideContent ? null : children}
-        {hideContent ? null : maybeText}
+        <View style={contentOpacityStyle}>
```
Comment (paullinator) [inline src/components/ui4/ButtonUi4.tsx:162]: This is a dangerous change! This is the type of high regression potential change I was referring to this week.

You are changing the way children are laid out in the button. This needs a full audit of everywhere ButtonUi4 is used. Note the difference before and after your change in the screenshots
Screenshot 2024-01-27 at 17 27 08

Screenshot 2024-01-27 at 17 27 22

The first image is not ideal but the issue is that it's different which it shouldn't be.

Change the View alignment style that affects children to exactly match what the LinearGradient had. Or better yet, punt this to 4.1 as it's not that urgent.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/ui4/ButtonUi4.tsx (hunk 1 above)
Comment (Jon-edge) [reply src/components/ui4/ButtonUi4.tsx:210]: This was a request by Madison

**Comment by Jon-edge** [comment]
Updated screens with Dev Tab additions

**Review by paullinator — APPROVED** [review]

**Pull request merged**